README.md

```diff
@@ -11,17 +11,25 @@ This project demonstrates an approach for building an IP connectivity-as-a-servi
 
 Some of the features and use cases that this project aims to support are:
 
 - **IoT networking** - connectivity to any node, anywhere
 - **Hybrid data center connectivity** - circumvents NAT challenges
 - **IP mobility** - /32 host routing allows addresses to be advertised anywhere with convergence times only limited by a round-trip time to a controller.
 - **Compliance** - Provide isolated connectivity among a set of nodes, even if they are running across multiple network administrative domains.
-- **Platform Agnostic** - Work independently from the infrastructure platform and support multiple operating systems (Linux, Mac, Windows).
-- **SaaS** - Built with a service-first mindset and provides enterprise auth out of the box
+- **Configurable Authentication** - Apex uses OpenID Connect (OIDC) for authentication, allowing you to choose from a broad ecosystem of OIDC providers. OIDC gateways are available to provide interoperability with other authentication schemes (i.e., LDAP).
+
+## Guiding Principles
+
+Our guiding principles help guide our decision-making. We use these principles when adding new features or making difficult decisions that require weighing different tradeoffs.
+
+1. **Service First** -- We intend for Apex to run as a service, whether a public service or one run internally by an organization. To ensure that we get this right, the Apex development team must also run it as a service. We are doing this on top of [operate-first](https://www.operate-first.cloud/) community infrastructure and by following [open-source services](https://www.operate-first.cloud/community/open-source-services.html) principles: http://apex.apps.smaug.na.operate-first.cloud/.
+2. **Simple UX Above Features** -- Networking technology is often made incredibly complex. One key value that the Apex service can provide is a simplified user experience for a much more challenging problem. When deciding how or if to add a feature, we value retaining the simplicity of the user experience over functionality.
+3. **Optimized Connectivity** -- Forcing traffic through a central hub or other intermediary does not provide ideal network connectivity. Instead, we will implement various techniques to provide direct, mesh connectivity wherever possible, even in places where NAT or firewalls would typically prevent it.
+4. **Platform Agnostic** - Apex will work independently from the infrastructure platform and support multiple operating systems (Linux, Mac, Windows). We value making the service easy to use with different infrastructure platforms (Kubernetes, for example), but we will avoid changes that tie the service to any particular platform.
 
```

Comment:
@russellb looks good to me. Thanks for putting this together.
"Zero Trust Security" should also be a core principle ? Zero Touch Device Provisioning/Onboarding (Seamless onboarding) probably the other one, but not sure.

Reply:
+1 this is great, ty! I agree with Anil on a security bullet if it doesn't overload it. I could see it being generic sec bullet with mentions of ZTN and even SASE. Could potentially wedge in observability in that bullet as well.

Reply:
Good point about security. I think that one is worth adding.  How about this:

---

5. **Secure Traffic** -- All traffic must be encrypted and the service should never have the ability to decrypt it. Even in the case that traffic needs to be relayed to provide connectivity, the relay must not be able to decrypt the traffic it is relaying[^1].

[^1]: See [Issue #169](https://github.com/redhat-et/apex/issues/169). Our first forwarding implementation is by using a normal apex node on the network which decrypts, makes an IP routing decision, and sends it back out over another encrypted tunnel.

Reply:
I wanted to capture the aspirational goal here, but also acknowledge that we haven't fully achieved it yet.

Reply:
I would go further and add a bullet that says something like "Secure By Design" or somesuch...

This is my take, but it's too many bullets - probably should live in a `docs/security.md` or somesuch.

5. Zero Trust Architecture -- The service has been designed with the tenets of Zero Trust Networking in mind.
- Continuous monitoring and validation: Devices check in to the service regularly for peer updates. User logins are managed via OIDC which allows the use of short-lived access tokens that can be periodically refreshed. In effect, both Devices and Users check in regularly and their Peers or Tokens can be quickly revoked if needed.
- Device Authorization: Our device on-boarding process associates devices with authenticated users.
- Least Privilege: Users are assigned only the privileges they require (add footnote about oauth scopes)
- Microsegmentation: Users are broken into smaller segments using Zones and may only communicate within the same zone

The points about encryption and traffic security are good ones! I also feel like that needs to be in a dedicated security doc.

.... so maybe that's a long way of saying the bullet should just be that Security is a guiding principle.

Reply:
Thanks, everyone.

I added a generic security bullet that refers to a dedicated security document. The security document includes my bullet on encrypted traffic, as well as all of the ones proposed by @dave-tucker.